labs/203-custom-userspace-instrumentation-in-c/README.md

```diff
@@ -0,0 +1,283 @@
+## Custom Userspace Instrumentation of C/C++ Applications
+
+In this lab, you will learn how to instrument, compile and trace your own application using LTTng userspace instrumentation. We'll instrument a simple C MPI application doing message passing between threads.
+
+*Pre-requisites*: Have Trace Compass installed and opened. Have lttng-ust installed, most notably the `liblttng-ust-dev` package. You can follow the [Install LTTng on Ubuntu](../001-instal-lttng-on-ubuntu/) lab or read the [LTTng web site](http://lttng.org) for more information.
+
+- - -
+
+### Task 1: Understand The MPI Application To Trace
+
+For this lab, we'll instrument an MPI application, taken from the [MPI tutorial](https://mpitutorial.com/) examples. We'll instrument the `ring` application, in which one mpi worker sends a value to the next and so on until the value comes back to the first worker. The subset of files used for this lab, original and instrumented, are in the [code](code/) directory of this lab. To compile and run it, you'll need the `openmpi` package (`sudo apt-get install openmpi`). A trace of the application is also available in the trace archive that comes with this tutorial.
```

Comment:
The package on Ubuntu/Debian to get the openmpi compiler, libraries and runtime is 'mpi-default-dev'.